modules/basegl/src/processors/imageprocessing/imagesubsetgl.cpp

```diff
@@ -0,0 +1,239 @@
+        util::makeBuffer<vec2>({{-1.0f, -1.0f}, {1.0f, -1.0f}, {-1.0f, 1.0f}, {1.0f, 1.0f}});
+    vec2 texOffset = vec2(0.f) / vec2(dims_);
+    vec2 texEnd = vec2(1.f) / vec2(dims_);
+    texCoordsBuffer_ = util::makeBuffer<vec2>(
+        {texOffset, {texEnd.x, texOffset.y}, {texOffset.x, texEnd.y}, texEnd});
+    auto indices = util::makeIndexBuffer({0, 1, 2, 3});
+    rect_.addBuffer(BufferType::PositionAttrib, verticesBuffer);
+    rect_.addBuffer(BufferType::TexcoordAttrib, texCoordsBuffer_);
+    rect_.addIndicies(Mesh::MeshInfo(DrawType::Triangles, ConnectivityType::Strip), indices);
+}
+
+ImageSubsetGL::~ImageSubsetGL() = default;
+
+void ImageSubsetGL::process() {
+    const ivec2 offset{rangeX_.get().x, rangeY_.get().x};
+    const ivec2 dim = ivec2{rangeX_.get().y, rangeY_.get().y} - offset;
+    if (dim == ivec2(dims_))
+        outport_.setData(inport_.getData());
+    else {
+        auto image = inport_.getData();
+        auto outImage = std::make_shared<Image>(dim, image->getDataFormat());
+        // pass meta data on
+        outImage->copyMetaDataFrom(*image);
+
+        outport_.setData(outImage);
+
+        TextureUnit::setZeroUnit();
+
+        TextureUnit colorUnit, depthUnit, pickingUnit;
+        utilgl::activateAndClearTarget(outport_, ImageType::ColorDepthPicking);
+
+        shader_.activate();
+        shader_.setUniform("color_", colorUnit.getUnitNumber());
+        shader_.setUniform("depth_", depthUnit.getUnitNumber());
+        shader_.setUniform("picking_", pickingUnit.getUnitNumber());
+
+        utilgl::bindTextures(*image, colorUnit, depthUnit, pickingUnit);
+        // Texture coordinates of the region we should copy
+        vec2 texOffset = vec2(offset) / vec2(dims_);
+        vec2 texEnd = vec2(offset + dim) / vec2(dims_);
+        auto coords = texCoordsBuffer_->getEditableRAMRepresentation();
+        coords->set(0, texOffset);
+        coords->set(1, {texEnd.x, texOffset.y});
+        coords->set(2, {texOffset.x, texEnd.y});
+        coords->set(3, texEnd);
+
+        utilgl::Enable<MeshGL> enable(rect_.getRepresentation<MeshGL>());
+        utilgl::DepthFuncState depth(GL_ALWAYS);
+        glDrawArrays(GL_TRIANGLE_STRIP, 0, 4);
+
+        shader_.deactivate();
+        utilgl::deactivateCurrentTarget();
+        TextureUnit::setZeroUnit();
+    }
+}
+
+void ImageSubsetGL::invokeEvent(Event* event) {
+    // Allow panning and prevent interaction events from propagating (does not make
+    switch (event->hash()) {
+        case MouseEvent::chash():
+            handleMousePan(static_cast<MouseEvent*>(event));
+            event->markAsUsed();
+            break;
+        case TouchEvent::chash():
+            handleTouchEvent(static_cast<TouchEvent*>(event));
+            event->markAsUsed();
+            break;
+        case WheelEvent::chash():
+        case GestureEvent::chash():
+            // GestureEvents are better handles using TouchEvent
+            event->markAsUsed();
```

Comment:
you mark events as used both here and in the `handleMousePan` and the `handleTouchEvent`, would be enough in those function, and we should only mark them as used when we have used them.

Reply:
In general I agree (also reason for comment a few lines above). I did not think that it made sence to have that kind of interaction in the subset (at least not in the scenario I used it for). 
If someone wants to use it for a different purpose it would be straight forward to introcude a "propagateInteractions" boolean. That is also the reason for keeping the markAsUsed in both functions.

Reply:
Added more comments